dotCMS/src/main/java/com/dotcms/security/apps/AppsUtil.java

```diff
@@ -0,0 +1,150 @@
+package com.dotcms.security.apps;
+
+import com.dotcms.rest.api.v1.DotObjectMapperProvider;
+import com.dotmarketing.exception.DotDataException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.io.ByteStreams;
+import com.liferay.util.Base64;
+import com.liferay.util.Encryptor;
+import com.liferay.util.EncryptorException;
+import java.io.BufferedReader;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.security.Key;
+import java.util.ArrayList;
+import java.util.List;
+import javax.crypto.Cipher;
+import org.apache.commons.io.input.CharSequenceInputStream;
+import org.apache.commons.lang3.ArrayUtils;
+
+public class AppsUtil {
+
+    private static final ObjectMapper mapper = DotObjectMapperProvider.getInstance().getDefaultObjectMapper();
+
+    /**
+     * Given the AppSecrets this will return a char array representing the deserialized json object.
+     * No strings are created in the transformation process.
+     * @param object
+     * @return
+     * @throws DotDataException
+     */
+    static char[] toJsonAsChars(final AppSecrets object) throws DotDataException {
+        try {
+            final byte [] bytes = mapper.writeValueAsBytes(object);
+            return bytesToCharArrayUTF(bytes);
+        } catch (IOException e) {
+            throw new DotDataException(e);
+        }
+    }
+
+    /**
+     * Takes a char array representation of a json secret and generates the domain object.
+     * No strings are created in the transformation process.
+     * @param chars
+     * @return The Secrets domain model.
+     * @throws DotDataException
+     */
+    static AppSecrets readJson(final char[] chars) throws DotDataException {
+        try {
+            final byte [] bytes = charsToBytesUTF(chars);
+            return mapper.readValue(bytes, AppSecrets.class);
+        } catch (IOException e) {
+            throw new DotDataException(e);
+        }
+    }
+
+    /**
+     * This method takes a byte array and converts its contents into a char array
+     * No String middle man is created.
+     * https://stackoverflow.com/questions/8881291/why-is-char-preferred-over-string-for-passwords
+     * @param bytes
+     * @return char array
+     */
+    static char[] bytesToCharArrayUTF(final byte[] bytes) throws IOException {
+        final List<Integer> integers = new ArrayList<>(bytes.length);
+        try (final BufferedReader reader = new BufferedReader(
+                new InputStreamReader(new ByteArrayInputStream(bytes), StandardCharsets.UTF_8))) {
+
+            int chr;
+            while ((chr = reader.read()) != -1) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid assignments in operands](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5329810)